helix-term/src/ui/editor.rs

```diff
@@ -416,18 +416,31 @@ impl EditorView {
         let mut last_line_indent_level = 0;
         let indent_style = theme.get("ui.virtual.indent-guide");
 
-        let draw_indent_guides = |indent_level, line, surface: &mut Surface| {
+        let mut hide_indent_level = 0u16;
+        let mut draw_indent_guides = |indent_level, line, surface: &mut Surface| {
             if !config.indent_guides.render {
                 return;
             }
 
             for i in 0..(indent_level / tab_width as u16) {
-                surface.set_string(
-                    viewport.x + (i * tab_width as u16) - offset.col as u16,
-                    viewport.y + line,
-                    &indent_guide_char,
-                    indent_style,
-                );
+                if i < hide_indent_level {
+                    continue;
+                }
+
+                let visual_x = i * tab_width as u16;
+                let out_of_bounds =
+                    visual_x < offset.col as u16 || visual_x >= viewport.width + offset.col as u16;
```

Comment:
This seems overly complicated to me, you should be able to calculate the starting indent level outside the for loop then `for i in starting_level..(indent_level / tab_width as u16)` to start rendering at the right offset.

Reply:
It's not about the starting level.. it's about stopping and not exceeding as I understand.
Possibly, once the right limit has been found, code can be further optimized to reuse this value.
As the guides are an optional feature, optimization should not bleed into normal codeflow.

Reply:
It's about the starting level. The code is rendering the right split's indent guides, but since the starting column is further in the file it's rendering them off screen into the previous split.